manager/src/main/java/org/openremote/manager/security/KeyStoreServiceImpl.java

```diff
@@ -0,0 +1,194 @@
+
+	private final String OR_SSL_CLIENT_KEYSTORE_PASSWORD = "OR_SSL_CLIENT_KEYSTORE_PASSWORD";
+	private final String OR_SSL_CLIENT_TRUSTSTORE_PASSWORD = "OR_SSL_CLIENT_TRUSTSTORE_PASSWORD";
+
+	private final String OR_KEYSTORE_PASSWORD = "OR_KEYSTORE_PASSWORD";
+
+
+	private Path keyStorePath;
+	private Path trustStorePath;
+
+
+	public String OR_KEYSTORE_PASSWORD_DEFAULT;
+	public String keyStorePassword;
+
+	@Override
+	public int getPriority() {
+		return ManagerIdentityService.PRIORITY + 10;
+	}
+
+	@Override
+	public void init(Container container) throws Exception {
+		persistenceService = container.getService(PersistenceService.class);
+		identityService = container.getService(ManagerIdentityService.class);
+
+		OR_KEYSTORE_PASSWORD_DEFAULT = getString(container.getConfig(), "OR_ADMIN_PASSWORD", "secret");
+
+		keyStorePassword = getString(container.getConfig(), OR_KEYSTORE_PASSWORD, OR_KEYSTORE_PASSWORD_DEFAULT);
+	}
+
+	@Override
+	public void start(Container container) throws Exception {
+
+		String keyStoreEnv = getString(container.getConfig(), OR_SSL_CLIENT_KEYSTORE_FILE, null);
+		String trustStoreEnv = getString(container.getConfig(), OR_SSL_CLIENT_TRUSTSTORE_FILE, null);
+
+		Optional<Path> keyStorePath = keyStoreEnv != null ? Optional.of(Paths.get(keyStoreEnv)) : Optional.empty();
+		Optional<Path> trustStorePath = trustStoreEnv != null ? Optional.of(Paths.get(trustStoreEnv)) : Optional.empty();
+
+		String keyStorePassword = getString(container.getConfig(), OR_SSL_CLIENT_KEYSTORE_PASSWORD, String.valueOf(getKeyStorePassword()));
+		String trustStorePassword = getString(container.getConfig(), OR_SSL_CLIENT_TRUSTSTORE_PASSWORD, String.valueOf(getKeyStorePassword()));
+
+
+		if(keyStorePath.isPresent()){
+			this.keyStore = KeyStore.getInstance(keyStorePath.get().toFile(), keyStorePassword.toCharArray());
+		}else{
+			Path defaultKeyStorePath = persistenceService.resolvePath(Paths.get("keystores").resolve("client_keystore.p12"));
+			if(new File(defaultKeyStorePath.toUri()).exists()) {
+				this.keyStorePath = defaultKeyStorePath;
+				this.keyStore = KeyStore.getInstance(new File(defaultKeyStorePath.toUri()), getKeyStorePassword());
+			}else{
+				this.keyStore = createKeyStore(defaultKeyStorePath);
+			}
+
+			this.keyStorePath = defaultKeyStorePath;
+		}
+
+		if(trustStorePath.isPresent()){
+			this.keyStore = KeyStore.getInstance(trustStorePath.get().toFile(), trustStorePassword.toCharArray());
+		}else{
+			Path defaultTrustStorePath = persistenceService.resolvePath(Paths.get("keystores").resolve("client_truststore.p12"));
+			if(new File(defaultTrustStorePath.toUri()).exists()) {
+				this.trustStorePath = defaultTrustStorePath;
+				this.trustStore = KeyStore.getInstance(new File(defaultTrustStorePath.toUri()), getKeyStorePassword());
+			}else{
+				this.trustStore = createKeyStore(defaultTrustStorePath);
+			}
+			this.trustStorePath = defaultTrustStorePath;
+		}
+	}
```

Comment:
Any assignment of a custom truststore should probably be set as the default for the whole JVM